src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -375,3 +376,33 @@ void CoreWrap::add_extension(const Napi::CallbackInfo& info) {
         reportError(info.Env(), err.what());
     }
 }
+
+Napi::Value CoreWrap::query_model(const Napi::CallbackInfo& info) {
+    try {
+        if (info.Length() < 2 || !info[0].IsObject() || !info[1].IsString()) {
```

Comment:
You should check if info[0] contain model object like this:
`
const auto& model_prototype = info.Env().GetInstanceData<AddonData>()->model;
 if (model_prototype && model.InstanceOf(model_prototype.Value().As<Napi::Function>())) { }
`
The exact form of this condition I leave to you ;)